metricflow-semantics/metricflow_semantics/specs/patterns/entity_link_pattern.py

```diff
@@ -112,6 +111,19 @@ def _match_entity_links(self, candidate_specs: Sequence[LinkableInstanceSpec]) -
         shortest_entity_link_length = min(len(matching_spec.entity_links) for matching_spec in matching_specs)
         return tuple(spec for spec in matching_specs if len(spec.entity_links) == shortest_entity_link_length)
 
+    def _match_time_granularities(
```

Comment:
My original idea was to avoid all of this by converting to the ExpandedTimeGranularity as early as possible, but it looks like you're keeping them as strings until the resolver does its thing, which also seems reasonable to me.